### ISSUE TITLE
Overfitting and optimization

### ISSUE THREAD

**Issue by andac-demir** [rusty1s/pytorch_geometric#2008]
I experienced overfitting and slow optimization in training my datasets utilizing some of your GCN convolutions.
Then I also run some of your experimental scripts for sanity checks. One of them was: [proteins_diff_pool](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/proteins_diff_pool.py). I still observe overfitting in training and slow optimization. Are GNN's inherently tended to overfit and converge slowly? Thank you for your suggestions!

**Comment by rusty1s** [comment]
In general, GNNs tend to overfit quite heavily, especially in small datasets. People usually try to prevent this by making heavy usage of dropout after each GNN layer). I never witnessed  slow convergence by myself though (usually, the GNN will converge rather quickly and then just jump a little around the local minima). If you notice slow convergence, I suggest you add batch normalization after all your GNN layers.
